tlslite/tlsconnection.py

```diff
@@ -4090,14 +4134,12 @@ def _sigHashesToList(settings, privateKey=None, certList=None,
 
                 # in TLS 1.3 ECDSA key curve is bound to hash
                 if publicKey and version > (3, 3):
-                    size = len(publicKey)
-                    size, r = divmod(size, 8)
-                    size += int(bool(r))
-                    if size == 32 and hashName != "sha256":
+                    curve = publicKey.curve_name
+                    if curve == "NIST256p" and hashName != "sha256":
```

Comment:
That test seems to be duplicated in `_clientTLS13Handshake`

Reply:
true, fixed